src/Storages/System/StorageSystemRemoteDataPaths.cpp

```diff
@@ -59,6 +60,18 @@ Pipe StorageSystemRemoteDataPaths::read(
             std::vector<IDisk::LocalPathWithObjectStoragePaths> remote_paths_by_local_path;
             disk->getRemotePathsRecursive("store", remote_paths_by_local_path);
             disk->getRemotePathsRecursive("data", remote_paths_by_local_path);
+            if (context->getSettingsRef().traverse_shadow_remote_data_paths)
+                disk->getRemotePathsRecursive(
+                    "shadow",
+                    remote_paths_by_local_path,
+                    [](const String & local_path)
+                    {
+                        // `shadow/{backup_name}/revision.txt` is not an object metadata file
```

Comment:
Because essentially this function parses all found metadata files and returns pairs (`remote_path`, `local_path`). But unfortunately `revision.txt` on this level doesn't have metadata format(it just contains number of revision) and the function falls with an exception like `unable to parse file`

Reply:
Ok, this actually our mistake - we should not allow direct data writes into local files for remote disks. we must always write metadata and put data on object storage. :(